frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -226,25 +228,41 @@ private void refreshModel(
 
     setModel(new GraphTableModel(repositoryGraph));
 
-    Set<String> skippedBranchNames = repositorySnapshot.getSkippedBranchNames();
-    if (skippedBranchNames.nonEmpty()) {
-      val notification = getSkippedBranchesNotification(repositorySnapshot, gitRepository);
-      VcsNotifier.getInstance(project).notify(notification);
-    }
+    if (!macheteFileIsOpened(macheteFilePath)) {
+      // notify if branch listed in machete file does not exist
+      Set<String> skippedBranchNames = repositorySnapshot.getSkippedBranchNames();
+      if (skippedBranchNames.nonEmpty()) {
+        val notification = getSkippedBranchesNotification(repositorySnapshot, gitRepository);
+        VcsNotifier.getInstance(project).notify(notification);
+      }
 
-    Set<String> duplicatedBranchNames = repositorySnapshot.getDuplicatedBranchNames();
-    if (duplicatedBranchNames.nonEmpty()) {
-      // This warning notification will not cover other error notifications (e.g. when rebase errors occur)
-      VcsNotifier.getInstance(project).notifyWarning(/* displayId */ null,
-          getString("string.GitMachete.EnhancedGraphTable.duplicated-branches-text"),
-          String.join(", ", duplicatedBranchNames));
+      // notify if branch name listed in machete file appears more than once
+      Set<String> duplicatedBranchNames = repositorySnapshot.getDuplicatedBranchNames();
+      if (duplicatedBranchNames.nonEmpty()) {
+        // This warning notification will not cover other error notifications (e.g. when rebase errors occur)
+        VcsNotifier.getInstance(project).notifyWarning(/* displayId */ null,
+            getString("string.GitMachete.EnhancedGraphTable.duplicated-branches-text"),
+            String.join(", ", duplicatedBranchNames));
+      }
     }
 
     repaint();
     revalidate();
     doOnUIThreadWhenReady.run();
   }
 
+  @UIEffect
+  private boolean macheteFileIsOpened(Path macheteFilePath) {
+    val fileEditorManager = FileEditorManager.getInstance(project);
+    val macheteVirtualFile = List.of(fileEditorManager.getOpenFiles())
+        .find(virtualFile -> virtualFile.getPath().equals(macheteFilePath.toString()));
+    if (macheteVirtualFile.isEmpty()) {
+      return false;
```

Comment:
such scenario should not happen in reality, but it is a way to safely call `get()` on the optional value